### ISSUE TITLE
Fixes needed in docker-compose.yml generation

### ISSUE THREAD

**Issue by moufmouf** [theaentmachine/aent-docker-compose#12]
Here are fixes needed in order to generate a valid docker-compose.yml:

Version is missing. Use 3.3:

```
version: "3.3"
```

The code generates a list of single values objets for environment (not valid):
```
    environment:
    - MYSQL_ROOT_PASSWORD: foobar
    - MYSQL_DATABASE: myapp
```

You should instead generate an object:

```
    environment:
      MYSQL_ROOT_PASSWORD: foobar
      MYSQL_DATABASE: myapp
```
Notice the missing "-".

Also, labels are generated as a list, but we really want an object (because it is easier to merge):

Instead of:
```
labels:
    - traefik.backend=phpmyadmin
    - traefik.frontend.rule=Host:phpmyadmin.localhost
    - traefik.port=80
```

Use:
```
labels:
    traefik.backend: phpmyadmin
    traefik.frontend.rule: "Host:phpmyadmin.localhost"
    traefik.port: 80
```

Finally, in "volumes":

```
services:
  mysql:
    volumes:
      - type: volume
        source: mysql-data
        target: /var/lib/mysql
        readOnly: false
```

Option "readOnly" does not exist. It's "read_only".

Note: I suspect there might be other issues like this one. An idea might be to test the validity of the generated file against docker-compose. There is a "docker-compose config" command that validates the generated file: https://stackoverflow.com/questions/29225972/validating-docker-compose-yml-file

**Comment by JinTotonic** [comment]
All fixed in [PR #13](https://github.com/theaentmachine/aent-docker-compose/pull/13) :smiley:

**Comment by moufmouf** [comment]
:+1:
